crates/router/src/core/payment_methods/cards.rs

```diff
@@ -4663,3 +4733,39 @@ pub async fn list_countries_currencies_for_connector_payment_method_util(
             .collect(),
     }
 }
+pub async fn delete_connector_mandate_metadata(
+    db: &dyn db::StorageInterface,
+    payment_mandate_reference: storage::PaymentsMandateReference,
+    merchant_id: &str,
+    pm: &diesel_models::PaymentMethod,
+    key_store: &domain::MerchantKeyStore,
+    deleted_at_time: time::PrimitiveDateTime,
+) -> errors::RouterResult<()> {
+    // check if connector name supports update and also if its a valid mca,ie, the mca is there in the hashmap
+    let mca_ids = get_all_mcas(
+        db,
+        key_store,
+        merchant_id,
+        Some(false),
+        &payment_mandate_reference,
+    )
+    .await?;
+    // TODO: filter on the basis of the configs, for the connectors that don't support revoke
+    let filtered_mca_ids = mca_ids
+        .into_iter()
+        .filter(|(mca_id, _)| payment_mandate_reference.contains_key(mca_id))
+        .collect::<HashMap<_, _>>();
+
+    delete_payment_method_task(
+        db,
+        &pm.payment_method_id,
+        filtered_mca_ids,
+        merchant_id.to_string(),
+        deleted_at_time,
+        pm.customer_id.clone(),
+    )
+    .await
+    .change_context(errors::ApiErrorResponse::InternalServerError)?;
```

Comment:
Add a context message with `attach_printable()` / `attach_printable_lazy()`.
We'd prefer to do that everywhere we're raising internal server errors.